src/Processors/Transforms/SquashingChunksTransform.h

```diff
@@ -28,6 +28,27 @@ class SquashingChunksTransform : public ExceptionKeepingTransform
     Chunk finish_chunk;
 };
 
+class SquashingChunksTransformForBalancing : public ExceptionKeepingTransform
+{
+public:
+    explicit SquashingChunksTransformForBalancing(
+        const Block & header, size_t min_block_size_rows, size_t min_block_size_bytes);
+
+    String getName() const override { return "SquashingTransform"; }
+
+    void work() override;
+
+protected:
+    void onConsume(Chunk chunk) override;
+    GenerateResult onGenerate() override;
+    void onFinish() override;
+
+private:
+    NewSquashingTransform squashing;
```

Comment:
not good choise for naming.
I sudgest remove all old squashing.
You provide us 3 transforms: PlanSquashig, ApplySquash and just Squash. The last one does all the steps inside one transform. That transform replase all existed squashing transforms.